core/common/src/main/java/alluxio/underfs/ObjectUnderFileSystem.java

```diff
@@ -452,6 +452,19 @@ public UfsFileStatus getFileStatus(String path) throws IOException {
     }
   }
 
+  @Override
+  public UfsStatus getStatus(String path) throws IOException {
+    if (isRoot(path)) {
+      return getDirectoryStatus(path);
+    }
+
+    UfsFileStatus fileStatus = getFileStatus(path);
+    if (fileStatus != null) {
```

Comment:
I think getFileStatus will throw a FileNotFoundException instead of a null condition => need a try/catch

Reply:
Thanks for catching this! I updated this.